### ISSUE TITLE
Make the license irrevocable

### ISSUE THREAD

**Issue by JamesMcMahon** [jashkenas/lil-license#2]
From my [HackerNews comment](https://news.ycombinator.com/item?id=15293741#15294679).

I really wish it made clear that the license was irrevocable. I think the understanding in the community is that once you give permission for software to be used under a license you shouldn't go back on that, but other licenses, like Apache, make that explicit.

The Apache License 2.0 section:

>Subject to the terms and conditions of this License, each Contributor hereby grants to You a perpetual, worldwide, non-exclusive, no-charge, royalty-free, irrevocable copyright license to reproduce, prepare Derivative Works of, publicly display, publicly perform, sublicense, and distribute the Work and such Derivative Works in Source or Object form.

The GPL 3 section:

> All rights granted under this License are granted for the term of copyright on the Program, and are irrevocable provided the stated conditions are met.

**Comment by jashkenas** [comment]
I agree that the common community understanding is that you can't retract an open-source license after the fact.

And in addition, it appears as though OSS licenses like MIT that omit an explicit reference to revocability are also irrevocable:

https://opensource.stackexchange.com/questions/4012/are-licenses-irrevocable-by-default

> A non-exclusive copyright license (such as most FOSS licenses) can be revoked at any time only if there was no consideration involved. The United States Federal Circuit Court of Appeal took this on in Jacobsen v. Katzer in 2008 and ruled that there is consideration exchanged in the use of FOSS by a licensee. This indicates that an FOSS license that's silent on revocation is likely revocable only for violation of it's conditions.

In short, because you've given something of value under the license, you can't just take it back unilaterally.

Does that address your concern?

**Comment by JamesMcMahon** [comment]
Frankly no.

That stackoveflow links refers to a court case and legal precedent being interpreted by someone who is not acting as a lawyer.

1. I would get an actual lawyer to interpret the ruling.
2. Legal precedent can be overturned.

If the goal is a license that is simple and obvious why not explicitly state this?

**Comment by jashkenas** [comment]
We should certainly include a discussion about revocability with the lawyers who review the next draft of this, and I'm more than happy to explicitly include "irrevocable" if revocability would otherwise be the default — as this firm thinks it is:

https://www.sidley.com/en/insights/newsupdates/2013/02/the-terms-revocable-and-irrevocable-in-license-agreements-tips-and-pitfalls

But the crux of the issue is whether something like MIT is revocable or not. If it is, than that's a serious problem that LIL can help improve upon. If it isn't, then we would be doing a disservice by including "irrevocable" in the text, and implying that licenses that don't include it somehow aren't. That's part of the point of arriving at something minimal.

**Comment by jashkenas** [comment]
One more nice argument for including explicit irrevocability: https://news.ycombinator.com/item?id=12567157

**Comment by JamesMcMahon** [comment]
> But the crux of the issue is whether something like MIT is revocable or not. If it is, than that's a serious problem that LIL can help improve upon. If it isn't, then we would be doing a disservice by including "irrevocable" in the text, and implying that licenses that don't include it somehow aren't. That's part of the point of arriving at something minimal.

Yeah I agree, sussing that out will be the hard part.

Is this issue tightly coupled to the [question of sub licensing](https://github.com/jashkenas/lil-license/issues/5)? That Hackernews article makes it sound like it might be.
